### PR TITLE
ListStore Cleanup: Post list DB query improvements

### PR DESCRIPTION
**Pull request by oguzkocer**
This PR improves two DB queries we are making to get the post list from `PostStore`.

**Getting posts by local or remote ids: `getPostsByLocalOrRemotePostIds`**

In #1193, just as a temporary solution, I used a very direct solution where I walk over the ids and make individual DB queries for each one. In this PR this is changed to a single query.

One important change I've made in this PR is to change the return type of this method from a `Map` with `LocalOrRemoteId` as its key to a simple list. I've spent quite a bit of time thinking about this and at the end decided to go with this, because:

1. Just because we query a post with either a local id or a remote id doesn't mean that the post doesn't have both ids. By mapping to a specific type of id, we are limiting its usage and might be conveying the wrong intention. (meaning the consumer might think that it'll only have one type of valid id) We could add another type to describe this situation, but it didn't feel like it's worth it, but I'd be curious to hear what other people think about it.
2. I feel that the matching from local or remote ids to posts is an implementation detail for a specific feature. When we return a list of posts for the given local and remote ids, it feels like the library is already doing its job.

**Querying just ids for local posts**

`PostSqlUtils.getLocalPostIdsForFilter` was querying all the columns just to map it to its `id`, so `.columns(PostModelTable.ID)` was added to make this query more efficient. Since there is no type conversation 😭 I've added couple comments about this to make sure it's not confused to be a full model.

_This PR is opened as a draft PR as we should merge #1193 and change the base branch before merging this. It's separated from #1193 as it's self contained and should be easier to understand and discuss here._

### CODE REVIEW

**Comment by oguzkocer** [comment]
The changes in this PR has been reflected in the WPAndroid draft PR https://github.com/wordpress-mobile/WordPress-Android/pull/9508 in this commit: https://github.com/wordpress-mobile/WordPress-Android/pull/9508/commits/04b191f13065e51a95d3dfb3a1e5dc7bb76813b9.

File: fluxc/src/main/java/org/wordpress/android/fluxc/persistence/PostSqlUtils.java [hunk 1]
```diff
@@ -132,6 +135,31 @@ public static int insertOrUpdatePostOverwritingLocalChanges(PostModel post) {
...
+        for (LocalOrRemoteId localOrRemoteId : localOrRemoteIds) {
+            if (localOrRemoteId instanceof LocalId) {
+                localIds.add(((LocalId) localOrRemoteId).getValue());
+            } else if (localOrRemoteId instanceof RemoteId) {
+                remoteIds.add(((RemoteId) localOrRemoteId).getValue());
+            }
+        }
+        ConditionClauseBuilder<SelectQuery<PostModel>> whereQuery =
+                WellSql.select(PostModel.class).where().equals(PostModelTable.LOCAL_SITE_ID, localSiteId);
+        if (!localIds.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (oguzkocer) [reply fluxc/src/main/java/org/wordpress/android/fluxc/persistence/PostSqlUtils.java]: Yes we do, nice catch!

[the remark this answers is not in the document]
Comment (oguzkocer) [reply fluxc/src/main/java/org/wordpress/android/fluxc/persistence/PostSqlUtils.java]: Fixed in a894d8e2f98cf42b2b6e3f4df98423bc0b936b3c and unit test added in 61bbeef3848f6932fad544fdfb363f8d672e81da. The unit test is not perfect, but at least it verifies the behavior. Unfortunately I couldn't find a great way to write that unit test in Java without spending too much time on it, so for now will leave it like that.

**Comment by oguzkocer** [comment]
@malinajirka If the #1193, please feel free to update the base branch to `feature/master-post-filters`, mark this PR ready for review and merge it if everything looks good.

**Comment by malinajirka** [comment]
LGTM, thanks!

**Pull request merged**